Intersect.Client/Core/Graphics.cs

```diff
@@ -144,12 +148,71 @@ public static void DrawIntro()
 
         public static void DrawMenu()
         {
-            var imageTex = sContentManager.GetTexture(
-                Framework.Content.TextureType.Gui, ClientConfiguration.Instance.MenuBackground
-            );
+            // Draw an animated background in the main menu.
+            if (ClientConfiguration.Instance.UseAnimatedMenuBg)
+            {
+                var animTex = Globals.ContentManager.GetTexture(
+                    Framework.Content.TextureType.Animation, ClientConfiguration.Instance.AnimatedMenuBgTexture
+                );
 
-            if (imageTex != null)
+                if (animTex == null)
+                {
+                    return;
+                }
+
+                var destRectangle = new FloatRect();
+                var textFrames = ClientConfiguration.Instance.AnimatedMenuBgFrameCount;
+                var animDelay = ClientConfiguration.Instance.AnimatedMenuBgDelay;
+
+                mainmenuBgAnimTime += 1;
```

Comment:
Does the background image change in speed when you change frame rate settings in the client settings menu? Curious since I don't see any sort of "Delta Time" game clock calculations. Also, doing the frame advancement in a way that's locked to game time will make it so that a user actually knows the unit of time they're entering into `AnimatedMenuBgDelay`.

This is just a problem I observed when I was adding my screen shake effect. For an example of a system that uses the game clock to calculate when to advance a transition, you can look at `Fade.cs` - that's what I copied off of lol

Reply:
will take a look at it!

Reply:
Hey, thanks for pointing this out, the timing for the animation frames should be working properly in milliseconds now ;)